FIX: Workaround for supabase template setup failures (migration error, permission error, Kong config error)

retoruto-carry opened the pull request:
## Changes

- Customized `migrate.sh` to avoid early connection failures.
- Added permission-fixing steps for initialization scripts.
- Fixed Kong startup failure due to config file ownership.

---

## ✨ Overview

This pull request addresses multiple issues encountered when setting up a Supabase environment using the official **template YAML** provided by the project.  
Without this fix, the initial setup process would fail at various stages.

## 📋 Problem Summary

- During the initial setup, the migration script attempts to connect to the `_supabase` database, which does not exist yet, and fails.
- Permission errors occur because scripts under `/docker-entrypoint-initdb.d` do not have the appropriate file permissions, causing them to be ignored.
- Kong fails to start because its configuration file cannot be accessed due to improper file ownership.

These issues prevented a successful `docker-compose up` deployment of Supabase.

## 🐛 Main Problems and Error Logs

### 1. Migration script connection error

**Details**  
The `migrate.sh` script tries to connect to the `_supabase` database at the very beginning, even though the database has not been created yet.

**Eerror log**

```bash
psql: error: connection to server at "127.0.0.1" (::1), port 5432 failed: FATAL:  database "_supabase" does not exist
```

In the original `migrate.sh`, the script was designed to connect to the database specified by the `$POSTGRES_DB` environment variable (e.g., `_supabase`) by default.  
It did not explicitly set the `-d` (database) option for `psql`, so it relied on the value of `$PGDATABASE`.

As a result, during the initial setup:
- The `_supabase` database had not been created yet.
- However, the script tried to connect to `$POSTGRES_DB` (`_supabase`),  


---

### 2. Migration script permission issues

**Details**  
Files under `/docker-entrypoint-initdb.d` (`*.sh`, `*.sql`) did not have the correct executable/readable permissions, causing them to be skipped by the entrypoint.

**Error log**

```bash
/usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/migrate.sh
```

---

### 3. Kong configuration file permission error

**Details**  
At startup, Kong tries to load `/usr/local/kong/kong.yml`, but it fails because the file is still owned by `root` and not accessible by the `kong` user.

**Error log**

```bash
Error: permission denied while opening file /usr/local/kong/kong.yml
```

---

## ✅ Changes Made

| Problem                          | What I Did                                                                                   |
|:----------------------------------|:---------------------------------------------------------------------------------------------|
| Migration connection error        | Customized `migrate.sh` to connect to the `postgres` database first when creating `_supabase`. |
| Script permission issues          | Set permissions to `755` for `.sh` files and `644` for `.sql` files at container startup.     |
| Kong configuration permission error | Changed ownership of `/usr/local/kong` to `kong:kong` before starting Kong.                   |

---

## 🛠️ About Migration Script Changes

To address the initial setup issue,  
I **overrode** the official `migrate.sh` script with a modified version.

- The modified script connects to the default `postgres` database first,  
  creates the `_supabase` database,  
  and **then** continues with the usual migration steps.
- This change successfully bypasses the "database does not exist" error.

However, **this is a temporary solution**:  
- If the official `migrate.sh` gets updated in future releases,  
  this customization **would require manual synchronization**.
- Ideally, it would be better to patch it in a more maintainable way (e.g., applying minimal diffs, using entrypoint hooks, etc.).

For now, I chose this approach to prioritize **getting the environment up and running reliably**.

---

## 📜 Notes

- This pull request is a **workaround**, not a final ideal solution.
- Future updates to the official template may conflict with this change, so further adjustments may become necessary.

---

## 🙏 Acknowledgment

Thank you very much for providing such a great project.  
I'm happy to contribute to improving the developer experience.  

This is my first time submitting a pull request to an open-source project.  
If there are any mistakes or better ways to contribute, I sincerely apologize and would greatly appreciate any guidance!

While this pull request may not be perfect, I would sincerely appreciate any help in refining or improving it if needed.

I truly appreciate how incredibly useful Coolify is — having a stable and reliable way to host Supabase would make a tremendous difference for me!

---


## Related Issues

- https://github.com/coollabsio/coolify/issues/4911
- https://github.com/coollabsio/coolify/issues/4665

retoruto-carry committed:
fix: supabase setup failure because of permisions and migration-script

djsisson commented:
@retoruto-carry please note for postgres, migrations are only ran if no database exists, otherwise they are always skipped, this is by design.

So if you already have a postgres db and manually update the template, it will fail as the migrations are skipped, (the _supasee db is already created in one of the migration files)

I don't think forcing the migrations to run is a good idea since this goes against postgres defaults.

If you need to update, make a new service, backup your old db, then restore into new one.


Lastly, coolify does tend to keep resetting perms on files and that was annoying, I think it should prob be changed to docker config, since you can specify perms as part of the mount.


***edit***

ok you aren't forcing the migrations, but i can't see where in the original script it is trying to connect to _supabase, since that is only used by the pooler and logs, neither of which are needed by the main postgres migrations itself, since `$POSTGRES_DB` is supposed to be what is used by postgres i.e `postgres` 

anything related to _supabase is not ran until after the _supabase db is created, so can you give more information on if this was a clean install? as i can't see any errors

retoruto-carry commented:
@djsisson thank you for your reply!!!

I'm not entirely sure if this qualifies as a "clean install," but here's what I did:

https://github.com/user-attachments/assets/4e4cb21f-2686-4489-803a-24d5c8351c4f


From the Coolify UI, I selected **Supabase** and clicked the **Deploy** button without changing any configuration.  
After a short while, the deployment fails with an error related to `_supabase` not existing.


*(Side note: during this process, an unclosable "Edit" modal appears. I'm not sure if this is a Coolify bug, but it disappears after a page reload. It might just be my environment.)*

<img width="725" alt="image" src="https://github.com/user-attachments/assets/ebc4e667-b742-4912-ba90-7ad5133be7aa" />

<img width="725" alt="image" src="https://github.com/user-attachments/assets/9ed71bfc-9421-4d7d-8b75-57713c9159f3" />

<img width="983" alt="image" src="https://github.com/user-attachments/assets/47b2c87a-7ea8-41da-a036-183c26e5f97d" />


---

After clicking **Deploy**, I received the following errors:

### 🔴 Supabase Db Log

```
2025-04-29T10:49:26.473918455Z The files belonging to this database system will be owned by user "postgres".
2025-04-29T10:49:26.473961942Z This user must also own the server process.
2025-04-29T10:49:26.473965303Z 
2025-04-29T10:49:26.474799944Z The database cluster will be initialized with locale "en_US.UTF-8".
2025-04-29T10:49:26.474823629Z The default database encoding has accordingly been set to "UTF8".
2025-04-29T10:49:26.474834845Z The default text search configuration will be set to "english".
2025-04-29T10:49:26.474838632Z 
2025-04-29T10:49:26.474841771Z Data page checksums are disabled.
2025-04-29T10:49:26.474845191Z 
2025-04-29T10:49:26.474848334Z fixing permissions on existing directory /var/lib/postgresql/data ... ok
2025-04-29T10:49:26.475446453Z creating subdirectories ... ok
2025-04-29T10:49:26.475497147Z selecting dynamic shared memory implementation ... posix
2025-04-29T10:49:26.494780888Z selecting default max_connections ... 100
2025-04-29T10:49:26.517918162Z selecting default shared_buffers ... 128MB
2025-04-29T10:49:26.534744316Z selecting default time zone ... Etc/UTC
2025-04-29T10:49:26.535523799Z creating configuration files ... ok
2025-04-29T10:49:26.668635686Z running bootstrap script ... ok
2025-04-29T10:49:27.080454091Z performing post-bootstrap initialization ... ok
2025-04-29T10:49:27.161374595Z syncing data to disk ... ok
2025-04-29T10:49:27.161407350Z initdb: warning: enabling "trust" authentication for local connections
2025-04-29T10:49:27.161409841Z 
2025-04-29T10:49:27.161412206Z 
2025-04-29T10:49:27.161414010Z Success. You can now start the database server using:
2025-04-29T10:49:27.161423317Z initdb: hint: You can change this by editing pg_hba.conf or using the option -A, or --auth-local and --auth-host, the next time you run initdb.
2025-04-29T10:49:27.161429889Z 
2025-04-29T10:49:27.161432121Z     /usr/bin/pg_ctl -D /var/lib/postgresql/data -l logfile start
2025-04-29T10:49:27.161434010Z 
2025-04-29T10:49:27.292952067Z waiting for server to start.... done
2025-04-29T10:49:27.292989963Z server started
2025-04-29T10:49:27.316685045Z 
2025-04-29T10:49:27.316727513Z /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/init-scripts
2025-04-29T10:49:27.316730704Z 
2025-04-29T10:49:27.316732702Z /usr/local/bin/docker-entrypoint.sh: running /docker-entrypoint-initdb.d/migrate.sh
2025-04-29T10:49:27.345854465Z DO
2025-04-29T10:49:27.346846953Z /docker-entrypoint-initdb.d/migrate.sh: running /docker-entrypoint-initdb.d/init-scripts/00-schema.sql
2025-04-29T10:49:27.356399937Z CREATE ROLE
2025-04-29T10:49:27.357830532Z REVOKE
2025-04-29T10:49:27.359652852Z CREATE SCHEMA
2025-04-29T10:49:27.362312013Z CREATE FUNCTION
2025-04-29T10:49:27.364002293Z REVOKE
2025-04-29T10:49:27.365299302Z GRANT
2025-04-29T10:49:27.365901939Z /docker-entrypoint-initdb.d/migrate.sh: running /docker-entrypoint-initdb.d/init-scripts/00000000000000-initial-schema.sql
2025-04-29T10:49:27.376161615Z CREATE PUBLICATION
2025-04-29T10:49:27.377325906Z ALTER ROLE
2025-04-29T10:49:27.379052872Z CREATE ROLE
2025-04-29T10:49:27.380115328Z CREATE ROLE
2025-04-29T10:49:27.381468308Z GRANT ROLE
2025-04-29T10:49:27.382628636Z CREATE SCHEMA
2025-04-29T10:49:27.388481102Z CREATE EXTENSION
2025-04-29T10:49:27.395914257Z CREATE EXTENSION
2025-04-29T10:49:27.399896291Z CREATE EXTENSION
2025-04-29T10:49:27.401070039Z CREATE ROLE
2025-04-29T10:49:27.402342405Z CREATE ROLE
2025-04-29T10:49:27.403598357Z CREATE ROLE
2025-04-29T10:49:27.405298563Z CREATE ROLE
2025-04-29T10:49:27.406489045Z GRANT ROLE
2025-04-29T10:49:27.408228963Z GRANT ROLE
2025-04-29T10:49:27.409320368Z GRANT ROLE
2025-04-29T10:49:27.411168589Z GRANT ROLE
2025-04-29T10:49:27.412259545Z GRANT
2025-04-29T10:49:27.413914624Z ALTER DEFAULT PRIVILEGES
2025-04-29T10:49:27.414991301Z ALTER DEFAULT PRIVILEGES
2025-04-29T10:49:27.415985882Z ALTER DEFAULT PRIVILEGES
2025-04-29T10:49:27.416911092Z GRANT
2025-04-29T10:49:27.418019760Z ALTER ROLE
2025-04-29T10:49:27.419141590Z ALTER DEFAULT PRIVILEGES
2025-04-29T10:49:27.420200424Z ALTER DEFAULT PRIVILEGES
2025-04-29T10:49:27.421333699Z ALTER DEFAULT PRIVILEGES
2025-04-29T10:49:27.422393224Z ALTER ROLE
2025-04-29T10:49:27.423681976Z ALTER ROLE
2025-04-29T10:49:27.424246897Z /docker-entrypoint-initdb.d/migrate.sh: running /docker-entrypoint-initdb.d/init-scripts/00000000000001-auth-schema.sql
2025-04-29T10:49:27.434982412Z CREATE SCHEMA
2025-04-29T10:49:27.456604119Z CREATE TABLE
2025-04-29T10:49:27.461932257Z CREATE INDEX
2025-04-29T10:49:27.467840608Z CREATE INDEX
2025-04-29T10:49:27.469034938Z COMMENT
2025-04-29T10:49:27.479128050Z CREATE TABLE
2025-04-29T10:49:27.484603915Z CREATE INDEX
2025-04-29T10:49:27.489095441Z CREATE INDEX
2025-04-29T10:49:27.496949322Z CREATE INDEX
2025-04-29T10:49:27.498170195Z COMMENT
2025-04-29T10:49:27.514604444Z CREATE TABLE
2025-04-29T10:49:27.515828443Z COMMENT
2025-04-29T10:49:27.523151642Z CREATE TABLE
2025-04-29T10:49:27.545173029Z CREATE INDEX
2025-04-29T10:49:27.547671493Z COMMENT
2025-04-29T10:49:27.555041452Z CREATE TABLE
2025-04-29T10:49:27.556548157Z COMMENT
2025-04-29T10:49:27.557992480Z INSERT 0 7
2025-04-29T10:49:27.560121470Z CREATE FUNCTION
2025-04-29T10:49:27.561354023Z CREATE FUNCTION
2025-04-29T10:49:27.562491395Z CREATE FUNCTION
2025-04-29T10:49:27.563476438Z GRANT
2025-04-29T10:49:27.564506184Z CREATE ROLE
2025-04-29T10:49:27.565566196Z GRANT
2025-04-29T10:49:27.567151989Z GRANT
2025-04-29T10:49:27.568196522Z GRANT
2025-04-29T10:49:27.569162740Z ALTER ROLE
2025-04-29T10:49:27.570549087Z ALTER TABLE
2025-04-29T10:49:27.572465986Z ALTER TABLE
2025-04-29T10:49:27.573741254Z ALTER TABLE
2025-04-29T10:49:27.574932742Z ALTER TABLE
2025-04-29T10:49:27.576940319Z ALTER TABLE
2025-04-29T10:49:27.577625981Z /docker-entrypoint-initdb.d/migrate.sh: running /docker-entrypoint-initdb.d/init-scripts/00000000000002-storage-schema.sql
2025-04-29T10:49:27.586956736Z CREATE SCHEMA
2025-04-29T10:49:27.588360300Z GRANT
2025-04-29T10:49:27.590038161Z ALTER DEFAULT PRIVILEGES
2025-04-29T10:49:27.591578088Z ALTER DEFAULT PRIVILEGES
2025-04-29T10:49:27.592750527Z ALTER DEFAULT PRIVILEGES
2025-04-29T10:49:27.603193829Z CREATE TABLE
2025-04-29T10:49:27.608015233Z CREATE INDEX
2025-04-29T10:49:27.619235997Z CREATE TABLE
2025-04-29T10:49:27.625131597Z CREATE INDEX
2025-04-29T10:49:27.632261584Z CREATE INDEX
2025-04-29T10:49:27.633301536Z ALTER TABLE
2025-04-29T10:49:27.634850693Z CREATE FUNCTION
2025-04-29T10:49:27.636083996Z CREATE FUNCTION
2025-04-29T10:49:27.637382055Z CREATE FUNCTION
2025-04-29T10:49:27.639191067Z CREATE FUNCTION
2025-04-29T10:49:27.646520741Z CREATE TABLE
2025-04-29T10:49:27.647555679Z CREATE ROLE
2025-04-29T10:49:27.649033979Z GRANT
2025-04-29T10:49:27.652472952Z GRANT
2025-04-29T10:49:27.652617062Z GRANT
2025-04-29T10:49:27.653927707Z ALTER ROLE
2025-04-29T10:49:27.655323384Z ALTER TABLE
2025-04-29T10:49:27.656569906Z ALTER TABLE
2025-04-29T10:49:27.658008727Z ALTER TABLE
2025-04-29T10:49:27.659685651Z ALTER FUNCTION
2025-04-29T10:49:27.660732212Z ALTER FUNCTION
2025-04-29T10:49:27.661779045Z ALTER FUNCTION
2025-04-29T10:49:27.662958066Z ALTER FUNCTION
2025-04-29T10:49:27.663441435Z /docker-entrypoint-initdb.d/migrate.sh: running /docker-entrypoint-initdb.d/init-scripts/00000000000003-post-setup.sql
2025-04-29T10:49:27.672324641Z ALTER ROLE
2025-04-29T10:49:27.673490447Z ALTER ROLE
2025-04-29T10:49:27.675972389Z CREATE FUNCTION
2025-04-29T10:49:27.677459936Z CREATE EVENT TRIGGER
2025-04-29T10:49:27.683525958Z COMMENT
2025-04-29T10:49:27.685751350Z CREATE FUNCTION
2025-04-29T10:49:27.687096363Z COMMENT
2025-04-29T10:49:27.689011168Z DO
2025-04-29T10:49:27.690138385Z CREATE ROLE
2025-04-29T10:49:27.691211179Z GRANT
2025-04-29T10:49:27.692346680Z GRANT
2025-04-29T10:49:27.693616469Z GRANT
2025-04-29T10:49:27.694772564Z GRANT
2025-04-29T10:49:27.696321376Z GRANT
2025-04-29T10:49:27.696520241Z GRANT
2025-04-29T10:49:27.698057511Z GRANT
2025-04-29T10:49:27.698482250Z GRANT
2025-04-29T10:49:27.698488570Z GRANT
2025-04-29T10:49:27.700318680Z GRANT
2025-04-29T10:49:27.703338338Z GRANT
2025-04-29T10:49:27.707041668Z GRANT
2025-04-29T10:49:27.707577782Z /docker-entrypoint-initdb.d/migrate.sh: running /docker-entrypoint-initdb.d/init-scripts/98-webhooks.sql
2025-04-29T10:49:27.716194931Z psql: error: /docker-entrypoint-initdb.d/init-scripts/98-webhooks.sql: Permission denied
2025-04-29T10:49:28.106617188Z 
2025-04-29T10:49:28.106639936Z PostgreSQL Database directory appears to contain a database; Skipping initialization
2025-04-29T10:49:28.106643773Z 
2025-04-29T10:49:34.490992301Z 172.18.0.6 2025-04-29 10:49:34.490 UTC [48] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:34.491025385Z 172.18.0.6 2025-04-29 10:49:34.490 UTC [49] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:35.885737837Z 172.18.0.6 2025-04-29 10:49:35.884 UTC [50] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:37.006577422Z 172.18.0.6 2025-04-29 10:49:37.006 UTC [51] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:39.022679045Z 172.18.0.6 2025-04-29 10:49:39.022 UTC [60] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:41.778738086Z 172.18.0.6 2025-04-29 10:49:41.778 UTC [61] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:48.390398114Z 172.18.0.6 2025-04-29 10:49:48.389 UTC [79] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:48.393829999Z 172.18.0.6 2025-04-29 10:49:48.390 UTC [82] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:48.393857810Z 172.18.0.6 2025-04-29 10:49:48.390 UTC [84] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:48.393875358Z 172.18.0.6 2025-04-29 10:49:48.390 UTC [81] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:48.393878824Z 172.18.0.6 2025-04-29 10:49:48.391 UTC [78] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:48.393882086Z 172.18.0.6 2025-04-29 10:49:48.391 UTC [86] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:48.393885270Z 172.18.0.6 2025-04-29 10:49:48.393 UTC [87] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:48.393888510Z 172.18.0.6 2025-04-29 10:49:48.393 UTC [85] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:48.395638956Z 172.18.0.6 2025-04-29 10:49:48.395 UTC [83] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:48.395830389Z 172.18.0.6 2025-04-29 10:49:48.395 UTC [80] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:48.409502456Z 172.18.0.6 2025-04-29 10:49:48.409 UTC [88] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:51.339060869Z 172.18.0.6 2025-04-29 10:49:51.338 UTC [90] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:51.339111528Z 172.18.0.6 2025-04-29 10:49:51.338 UTC [89] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:52.418394199Z 172.18.0.6 2025-04-29 10:49:52.418 UTC [91] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:52.829335021Z 172.18.0.6 2025-04-29 10:49:52.829 UTC [92] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:53.575491197Z 172.18.0.6 2025-04-29 10:49:53.575 UTC [101] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
2025-04-29T10:49:54.978279497Z 172.18.0.6 2025-04-29 10:49:54.977 UTC [102] supabase_admin@_supabase FATAL:  database "_supabase" does not exist
```

### 🔴 Supabase Analytics Log

```
2025-04-29T11:21:01.924389228Z 
2025-04-29T11:21:02.116692784Z Crash dump is being written to: erl_crash.dump...done
2025-04-29T11:21:02.591927772Z LOGFLARE_NODE_HOST is: 127.0.0.1
2025-04-29T11:21:03.240414801Z 
2025-04-29T11:21:03.240457003Z 11:21:03.236 [info] Starting migration
2025-04-29T11:21:03.576246594Z 
2025-04-29T11:21:03.576270706Z 11:21:03.575 [error] Postgrex.Protocol (#PID<0.162.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:03.576275282Z 
2025-04-29T11:21:03.576277902Z 11:21:03.576 [error] Postgrex.Protocol (#PID<0.163.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:04.676709658Z 
2025-04-29T11:21:04.676755965Z 11:21:04.676 [error] Postgrex.Protocol (#PID<0.162.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:04.760940186Z 
2025-04-29T11:21:04.760974417Z 11:21:04.760 [error] Postgrex.Protocol (#PID<0.163.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:07.472691075Z 
2025-04-29T11:21:07.472720929Z 11:21:07.472 [error] Postgrex.Protocol (#PID<0.162.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:07.876309831Z 
2025-04-29T11:21:07.876360593Z 11:21:07.876 [error] Postgrex.Protocol (#PID<0.163.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:14.148854879Z 
2025-04-29T11:21:14.148880701Z 11:21:14.148 [error] Postgrex.Protocol (#PID<0.162.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:14.448204793Z 
2025-04-29T11:21:14.448264282Z 11:21:14.447 [error] Could not create schema migrations table. This error usually happens due to the following:
2025-04-29T11:21:14.448267405Z 
2025-04-29T11:21:14.448269337Z   * The database does not exist
2025-04-29T11:21:14.448271485Z   * The "schema_migrations" table, which Ecto uses for managing
2025-04-29T11:21:14.448273998Z     migrations, was defined by another library
2025-04-29T11:21:14.448275882Z   * There is a deadlock while migrating (such as using concurrent
2025-04-29T11:21:14.448277857Z     indexes with a migration_lock)
2025-04-29T11:21:14.448279515Z 
2025-04-29T11:21:14.448281207Z To fix the first issue, run "mix ecto.create" for the desired MIX_ENV.
2025-04-29T11:21:14.448283068Z 
2025-04-29T11:21:14.448284600Z To address the second, you can run "mix ecto.drop" followed by
2025-04-29T11:21:14.448286766Z "mix ecto.create", both for the desired MIX_ENV. Alternatively you may
2025-04-29T11:21:14.448288714Z configure Ecto to use another table and/or repository for managing
2025-04-29T11:21:14.448290641Z migrations:
2025-04-29T11:21:14.448292407Z 
2025-04-29T11:21:14.448294269Z     config :logflare, Logflare.Repo,
2025-04-29T11:21:14.448295965Z       migration_source: "some_other_table_for_schema_migrations",
2025-04-29T11:21:14.448297917Z       migration_repo: AnotherRepoForSchemaMigrations
2025-04-29T11:21:14.448299651Z 
2025-04-29T11:21:14.448301281Z The full error report is shown below.
2025-04-29T11:21:14.448303067Z 
2025-04-29T11:21:14.451837362Z ** (DBConnection.ConnectionError) connection not available and request was dropped from queue after 10954ms. This means requests are coming in and your connection pool cannot serve them fast enough. You can address this by:
2025-04-29T11:21:14.451862001Z 
2025-04-29T11:21:14.451865065Z   1. Ensuring your database is available and that you can connect to it
2025-04-29T11:21:14.451867570Z   2. Tracking down slow queries and making sure they are running fast enough
2025-04-29T11:21:14.451881989Z   3. Increasing the pool_size (although this increases resource consumption)
2025-04-29T11:21:14.451884012Z   4. Allowing requests to wait longer by increasing :queue_target and :queue_interval
2025-04-29T11:21:14.451885906Z 
2025-04-29T11:21:14.451887597Z See DBConnection.start_link/2 for more information
2025-04-29T11:21:14.451889374Z 
2025-04-29T11:21:14.451890982Z     (ecto_sql 3.10.1) lib/ecto/adapters/sql.ex:913: Ecto.Adapters.SQL.raise_sql_call_error/1
2025-04-29T11:21:14.451892923Z     (elixir 1.14.4) lib/enum.ex:1658: Enum."-map/2-lists^map/1-0-"/2
2025-04-29T11:21:14.451894866Z     (ecto_sql 3.10.1) lib/ecto/adapters/sql.ex:1005: Ecto.Adapters.SQL.execute_ddl/4
2025-04-29T11:21:14.451896694Z     (ecto_sql 3.10.1) lib/ecto/migrator.ex:738: Ecto.Migrator.verbose_schema_migration/3
2025-04-29T11:21:14.451898500Z     (ecto_sql 3.10.1) lib/ecto/migrator.ex:552: Ecto.Migrator.lock_for_migrations/4
2025-04-29T11:21:14.451900423Z     (ecto_sql 3.10.1) lib/ecto/migrator.ex:428: Ecto.Migrator.run/4
2025-04-29T11:21:14.451902653Z     (ecto_sql 3.10.1) lib/ecto/migrator.ex:170: Ecto.Migrator.with_repo/3
2025-04-29T11:21:14.451904619Z     nofile:1: (file)
2025-04-29T11:21:17.826280562Z 
2025-04-29T11:21:17.826321712Z 11:21:17.825 [info] Elixir.Logflare.SigtermHandler is being initialized...
2025-04-29T11:21:17.865001254Z 
2025-04-29T11:21:17.865026474Z 11:21:17.862 [error] Postgrex.Protocol (#PID<0.4726.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:17.865032363Z 
2025-04-29T11:21:17.865036516Z 11:21:17.863 [error] Postgrex.Protocol (#PID<0.4720.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:17.865041071Z 
2025-04-29T11:21:17.865044555Z 11:21:17.863 [error] Postgrex.Protocol (#PID<0.4723.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:17.865048888Z 
2025-04-29T11:21:17.865052177Z 11:21:17.863 [error] Postgrex.Protocol (#PID<0.4725.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:17.865056778Z 
2025-04-29T11:21:17.865060107Z 11:21:17.863 [error] Postgrex.Protocol (#PID<0.4728.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:17.865064128Z 
2025-04-29T11:21:17.865067503Z 11:21:17.863 [error] Postgrex.Protocol (#PID<0.4727.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:17.865071777Z 
2025-04-29T11:21:17.865075081Z 11:21:17.863 [error] Postgrex.Protocol (#PID<0.4719.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:17.865160784Z 
2025-04-29T11:21:17.865168926Z 11:21:17.863 [error] Postgrex.Protocol (#PID<0.4724.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:17.865173681Z 
2025-04-29T11:21:17.865177158Z 11:21:17.864 [error] Postgrex.Protocol (#PID<0.4729.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:17.868463645Z 
2025-04-29T11:21:17.868490950Z 11:21:17.868 [error] Postgrex.Protocol (#PID<0.4730.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:17.885066674Z 
2025-04-29T11:21:17.885123961Z 11:21:17.884 [notice] Application logflare exited: Logflare.Application.start(:normal, []) returned an error: shutdown: failed to start child: Cainophile.Adapters.Postgres
2025-04-29T11:21:17.885132407Z     ** (EXIT) {:error, :fatal, "3D000", :invalid_catalog_name, "database \"_supabase\" does not exist", [file: "postinit.c", line: "945", routine: "InitPostgres", severity: "FATAL"]}
2025-04-29T11:21:19.475504135Z {"Kernel pid terminated",application_controller,"{application_start_failure,logflare,{{shutdown,{failed_to_start_child,'Elixir.Cainophile.Adapters.Postgres',{error,fatal,<<\"3D000\">>,invalid_catalog_name,<<\"database \\"_supabase\\" does not exist\">>,[{file,<<\"postinit.c\">>},{line,<<\"945\">>},{routine,<<\"InitPostgres\">>},{severity,<<\"FATAL\">>}]}}},{'Elixir.Logflare.Application',start,[normal,[]]}}}"}
2025-04-29T11:21:19.475746118Z Kernel pid terminated (application_controller) ({application_start_failure,logflare,{{shutdown,{failed_to_start_child,'Elixir.Cainophile.Adapters.Postgres',{error,fatal,<<"3D000">>,invalid_catalog_name,<<"database \"_supabase\" does not exist">>,[{file,<<"postinit.c">>},{line,<<"945">>},{routine,<<"InitPostgres">>},{severity,<<"FATAL">>}]}}},{'Elixir.Logflare.Application',start,[normal,[]]}}})
2025-04-29T11:21:19.478751862Z 
2025-04-29T11:21:19.656057506Z Crash dump is being written to: erl_crash.dump...done
2025-04-29T11:21:20.109314176Z LOGFLARE_NODE_HOST is: 127.0.0.1
2025-04-29T11:21:20.681068582Z 
2025-04-29T11:21:20.681116428Z 11:21:20.676 [info] Starting migration
2025-04-29T11:21:21.033367985Z 
2025-04-29T11:21:21.033408518Z 11:21:21.033 [error] Postgrex.Protocol (#PID<0.163.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:21.033413708Z 
2025-04-29T11:21:21.033416560Z 11:21:21.033 [error] Postgrex.Protocol (#PID<0.162.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:22.397611679Z 
2025-04-29T11:21:22.397654158Z 11:21:22.397 [error] Postgrex.Protocol (#PID<0.162.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:22.824755147Z 
2025-04-29T11:21:22.824776070Z 11:21:22.824 [error] Postgrex.Protocol (#PID<0.163.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
2025-04-29T11:21:24.039400579Z 
2025-04-29T11:21:24.039441201Z 11:21:24.039 [error] Postgrex.Protocol (#PID<0.162.0>) failed to connect: ** (Postgrex.Error) FATAL 3D000 (invalid_catalog_name) database "_supabase" does not exist
Supabase Auth
Only Show Number of Lines
*
100
Refresh
Stream Logs
Include Timestamps

Refresh to get the logs...
Supabase Db
Only Show Number of Lines
*
100
Refresh
Stream Logs
Include Timestamps

Refresh to get the logs...
Supabase Edge Functions
Only Show Number of Lines
*
100
Refresh
Stream Logs
Include Timestamps

Refresh to get the logs...
Supabase Kong
Only Show Number of Lines
*
100
Refresh
Stream Logs
Include Timestamps

Refresh to get the logs...
Supabase Meta
Only Show Number of Lines
*
100
Refresh
Stream Logs
Include Timestamps

Refresh to get the logs...
Supabase Minio
Only Show Number of Lines
*
100
Refresh
Stream Logs
Include Timestamps

Refresh to get the logs...
Supabase Rest
Only Show Number of Lines
*
100
Refresh
Stream Logs
Include Timestamps

Refresh to get the logs...
Supabase Storage
Only Show Number of Lines
*
100
Refresh
Stream Logs
Include Timestamps

Refresh to get the logs...
Supabase Studio
Only Show Number of Lines
*
100
Refresh
Stream Logs
Include Timestamps

Refresh to get the logs...
Supabase Supavisor
Only Show Number of Lines
*
100
Refresh
Stream Logs
Include Timestamps

Refresh to get the logs...
Supabase Vector
Only Show Number of Lines
*
100
Refresh
Stream Logs
Include Timestamps

Refresh to get the logs...
```

_This error in Supabase Analytics is a consequence of the Supabase Db failing to initialize properly._

---

Then I manually stopped the Supabase Db container and entered it to inspect:




```sh
➜  coolify git:(fix/supabase-setup) ssh -i ~/.ssh/coolify.pem -p 32222 custom-user-name@xxx.xxx.xxx.xxx

coolify-manager@vm-0153f606-d3:~$ sudo docker ps
[sudo] password for coolify-manager:
CONTAINER ID   IMAGE                                                               COMMAND                  CREATED          STATUS                            PORTS                                                                                                                                                        NAMES
037b82b4851b   supabase/logflare:1.4.0                                             "sh run.sh"              20 minutes ago   Up 9 seconds (health: starting)                                                                                                                                                                supabase-analytics-e8sk4o8k0o8o40gswgogwsg0
5d4289f93d36   supabase/postgres:15.8.1.048                                        "docker-entrypoint.s…"   21 minutes ago   Up 20 minutes (healthy)           5432/tcp                                                                                                                                                     supabase-db-e8sk4o8k0o8o40gswgogwsg0
d01c2f9eaca6   minio/minio                                                         "/usr/bin/docker-ent…"   21 minutes ago   Up 20 minutes (healthy)           9000/tcp                                                                                                                                                     supabase-minio-e8sk4o8k0o8o40gswgogwsg0
e4ec22073e53   timberio/vector:0.28.1-alpine                                       "/usr/local/bin/vect…"   21 minutes ago   Up 20 minutes (healthy)                                                                                                                                                                        supabase-vector-e8sk4o8k0o8o40gswgogwsg0
67e15f552af1   darthsim/imgproxy:v3.8.0                                            "imgproxy"               21 minutes ago   Up 20 minutes (healthy)           8080/tcp                                                                                                                                                     imgproxy-e8sk4o8k0o8o40gswgogwsg0
0e16207ea071   ghcr.io/coollabsio/sentinel:0.0.15                                  "/app/sentinel"          11 hours ago     Up 11 hours (healthy)                                                                                                                                                                          coolify-sentinel
5a1945330046   mariadb:11                                                          "docker-entrypoint.s…"   2 days ago       Up 16 hours (healthy)             0.0.0.0:3306->3306/tcp, :::3306->3306/tcp                                                                                                                    dkkw8ggo0c08wsg0wgo000c8
67461b594a04   ead658eb8f49                                                        "docker-php-entrypoi…"   2 days ago       Up 16 hours                       80/tcp, 8080/tcp                                                                                                                                             nwggwkw8woks8ok8wo8o8scc-172114065921
b4819fef84b2   wogggcgsss44go0ok8s480gw:6b080dd197895c5b9d86ca795cd2e59372ea31f0   "docker-php-entrypoi…"   2 days ago       Up 16 hours                       80/tcp, 8080/tcp                                                                                                                                             wogggcgsss44go0ok8s480gw-172025814912
d59875346a79   mariadb:11                                                          "docker-entrypoint.s…"   2 days ago       Up 16 hours (healthy)             0.0.0.0:3307->3306/tcp, :::3307->3306/tcp                                                                                                                    g8gso4s8kk8c0ss8gowckcsk
165706cfeca3   lscr.io/linuxserver/phpmyadmin:latest                               "/init"                  2 days ago       Up 16 hours (healthy)             80/tcp, 443/tcp                                                                                                                                              phpmyadmin-xggks88o804k0gsgcso4w4ss
8bf1ea1151d7   redis:7.2                                                           "docker-entrypoint.s…"   2 days ago       Up 16 hours (healthy)             6379/tcp                                                                                                                                                     ugcwwk8c8k8gsws4wgowks8w
70795a7c0e5d   redis:7.2                                                           "docker-entrypoint.s…"   2 days ago       Up 16 hours (healthy)             6379/tcp                                                                                                                                                     j0ocs4s44k0ggcowocssk0wg
bc531d8ebf25   traefik:v3.1                                                        "/entrypoint.sh --pi…"   2 days ago       Up 16 hours (healthy)             0.0.0.0:80->80/tcp, :::80->80/tcp, 0.0.0.0:443->443/tcp, :::443->443/tcp, 0.0.0.0:8080->8080/tcp, :::8080->8080/tcp, 0.0.0.0:443->443/udp, :::443->443/udp   coolify-proxy
d586cfa4f107   ghcr.io/coollabsio/coolify:4.0.0-beta.409                           "docker-php-serversi…"   2 days ago       Up 16 hours (healthy)             8000/tcp, 8443/tcp, 9000/tcp, 0.0.0.0:8000->8080/tcp, :::8000->8080/tcp                                                                                      coolify
7194cc3d5314   redis:7-alpine                                                      "docker-entrypoint.s…"   2 days ago       Up 16 hours (healthy)             6379/tcp                                                                                                                                                     coolify-redis
3b8d697f131c   ghcr.io/coollabsio/coolify-realtime:1.0.6                           "/bin/sh /soketi-ent…"   2 days ago       Up 16 hours (healthy)             0.0.0.0:6001-6002->6001-6002/tcp, :::6001-6002->6001-6002/tcp                                                                                                coolify-realtime
2dedc4cd17d7   postgres:15-alpine                                                  "docker-entrypoint.s…"   2 days ago       Up 16 hours (healthy)             5432/tcp                                                                                                                                                     coolify-db
coolify-manager@vm-0153f606-d3:~$ sudo docker exec -it 5d4289f93d36 bash
root@5d4289f93d36:/# psql -U postgres
psql (15.8)
Type "help" for help.

postgres=# \pset pager off
Pager usage is off.
postgres=# \pset pager off
Pager usage is off.
postgres=# \l
                                                          List of databases
   Name    |     Owner      | Encoding |   Collate   |    Ctype    | ICU Locale | Locale Provider |         Access privileges
-----------+----------------+----------+-------------+-------------+------------+-----------------+-----------------------------------
 postgres  | postgres       | UTF8     | en_US.UTF-8 | en_US.UTF-8 |            | libc            | =Tc/postgres                     +
           |                |          |             |             |            |                 | postgres=CTc/postgres            +
           |                |          |             |             |            |                 | dashboard_user=CTc/postgres
 template0 | supabase_admin | UTF8     | en_US.UTF-8 | en_US.UTF-8 |            | libc            | =c/supabase_admin                +
           |                |          |             |             |            |                 | supabase_admin=CTc/supabase_admin
 template1 | supabase_admin | UTF8     | en_US.UTF-8 | en_US.UTF-8 |            | libc            | =c/supabase_admin                +
           |                |          |             |             |            |                 | supabase_admin=CTc/supabase_admin
(3 rows)
```

```sh
root@5d4289f93d36:/# ls -la /docker-entrypoint-initdb.d/
total 24
drwxr-xr-x 1 root root 4096 Mar  6 10:03 .
drwxr-xr-x 1 root root 4096 Apr 29 10:49 ..
drwxr-xr-x 1 root root 4096 Apr 29 10:49 init-scripts
-rwxr-xr-x 1 root root 3027 Mar  6 10:03 migrate.sh
drwxr-xr-x 1 root root 4096 Apr 29 10:49 migrations
root@5d4289f93d36:/# cat /docker-entrypoint-initdb.d/migrate.sh
#!/bin/sh
set -eu

#######################################
# Used by both ami and docker builds to initialise database schema.
# Env vars:
#   POSTGRES_DB        defaults to postgres
#   POSTGRES_HOST      defaults to localhost
#   POSTGRES_PORT      defaults to 5432
#   POSTGRES_PASSWORD  defaults to ""
#   USE_DBMATE         defaults to ""
# Exit code:
#   0 if migration succeeds, non-zero on error.
#######################################

export PGDATABASE="${POSTGRES_DB:-postgres}"
export PGHOST="${POSTGRES_HOST:-localhost}"
export PGPORT="${POSTGRES_PORT:-5432}"
export PGPASSWORD="${POSTGRES_PASSWORD:-}"

# if args are supplied, simply forward to dbmate
connect="$PGPASSWORD@$PGHOST:$PGPORT/$PGDATABASE?sslmode=disable"
if [ "$#" -ne 0 ]; then
    export DATABASE_URL="${DATABASE_URL:-postgres://supabase_admin:$connect}"
    exec dbmate "$@"
    exit 0
fi

db=$( cd -- "$( dirname -- "$0" )" > /dev/null 2>&1 && pwd )
if [ -z "${USE_DBMATE:-}" ]; then
    psql -v ON_ERROR_STOP=1 --no-password --no-psqlrc -U supabase_admin <<EOSQL
do \$\$
begin
  -- postgres role is pre-created during AMI build
  if not exists (select from pg_roles where rolname = 'postgres') then
    create role postgres superuser login password '$PGPASSWORD';
    alter database postgres owner to postgres;
  end if;
end \$\$
EOSQL
    # run init scripts as postgres user
    for sql in "$db"/init-scripts/*.sql; do
        echo "$0: running $sql"
        psql -v ON_ERROR_STOP=1 --no-password --no-psqlrc -U postgres -f "$sql"
    done
    psql -v ON_ERROR_STOP=1 --no-password --no-psqlrc -U postgres -c "ALTER USER supabase_admin WITH PASSWORD '$PGPASSWORD'"
    # run migrations as super user - postgres user demoted in post-setup
    for sql in "$db"/migrations/*.sql; do
        echo "$0: running $sql"
        psql -v ON_ERROR_STOP=1 --no-password --no-psqlrc -U supabase_admin -f "$sql"
    done
else
    psql -v ON_ERROR_STOP=1 --no-password --no-psqlrc -U supabase_admin <<EOSQL
  create role postgres superuser login password '$PGPASSWORD';
  alter database postgres owner to postgres;
EOSQL
    # run init scripts as postgres user
    DBMATE_MIGRATIONS_DIR="$db/init-scripts" DATABASE_URL="postgres://postgres:$connect" dbmate --no-dump-schema migrate
    psql -v ON_ERROR_STOP=1 --no-password --no-psqlrc -U postgres -c "ALTER USER supabase_admin WITH PASSWORD '$PGPASSWORD'"
    # run migrations as super user - postgres user demoted in post-setup
    DBMATE_MIGRATIONS_DIR="$db/migrations" DATABASE_URL="postgres://supabase_admin:$connect" dbmate --no-dump-schema migrate
fi

# run any post migration script to update role passwords
postinit="/etc/postgresql.schema.sql"
if [ -e "$postinit" ]; then
    echo "$0: running $postinit"
    psql -v ON_ERROR_STOP=1 --no-password --no-psqlrc -U supabase_admin -f "$postinit"
fi

# once done with everything, reset stats from init
psql -v ON_ERROR_STOP=1 --no-password --no-psqlrc -U supabase_admin -c 'SELECT extensions.pg_stat_statements_reset(); SELECT pg_stat_reset();' || true
```

### ❗ Environment Assumption

In the official Coolify Supabase template, the following environment variable is defined in advance:

```sh
POSTGRES_DB=_supabase
```

Therefore, inside the default `migrate.sh`, this line:

```sh
export PGDATABASE="${POSTGRES_DB:-postgres}"
```

will result in `PGDATABASE=_supabase`.

However, the `_supabase` database **does not yet exist at this point**, because it’s supposed to be created **later in the migration file** (`97-_supabase.sql`).

This causes a critical ordering issue.

---

### ⚠️ How the Problem Occurs

1. `PGDATABASE` is set to `_supabase`.

```sh
export PGDATABASE="${POSTGRES_DB:-postgres}"
```

2. If dbmate is **not** used, the script takes the standard psql path and runs:

```sh
for sql in "$db"/init-scripts/*.sql; do
    echo "$0: running init script $sql"
    psql -v ON_ERROR_STOP=1 --no-password --no-psqlrc -U postgres -f "$sql"
done
```

✅ This part is fine because it connects to `postgres` explicitly.

3. Then, the script proceeds to run all `migrations/*.sql` with:

```sh
for sql in "$db"/migrations/*.sql; do
    echo "$0: running migration $sql"
    psql -v ON_ERROR_STOP=1 --no-password --no-psqlrc -U supabase_admin -f "$sql"
done
```

⚠️ Here, the connection target is `$PGDATABASE`, which is `_supabase`.

4. If the first file is `97-_supabase.sql`, it **should** create the `_supabase` database.  
   However, since the script is **already trying to connect to `_supabase`**, it fails:

```sh
psql: error: connection to server at "127.0.0.1", port 5432 failed: FATAL: database "_supabase" does not exist
```

---

### ✅ Solution

Only `97-_supabase.sql` should be executed by connecting to the default `postgres` database:

```sh
psql -U supabase_admin -d postgres -f "$db/migrations/97-_supabase.sql"
```

Then, switch to `_supabase` and run the rest of the migration files as intended.

---

### 🔧 How I Fixed It

I modified the `migrate.sh` script to:

- Run `97-_supabase.sql` first with `-d postgres`
- Then run the rest of the migrations with the default `$PGDATABASE` (`_supabase`)

This ensures the correct creation order and prevents early connection attempts to a non-existent database.

While this is a temporary workaround and may not be ideal (especially as upstream `migrate.sh` changes could cause drift), it allowed me to deploy Supabase via Coolify cleanly.

djsisson commented:
@retoruto-carry there must be some kind of permission issue with your files, i just did what you did, it worked fine, including kong

![image](https://github.com/user-attachments/assets/4a75dde1-7c3f-49b6-82a6-24aaafe2bd00)

-

![image](https://github.com/user-attachments/assets/b9c59f2a-bf13-4bbe-bf49-6a4bdc7e8620)


there is nothing wrong with the current migration file, it does not access the database _supabase, it is only from the 3 volume mounted scripts, which in your case were failed to be read, which is why _supabase was failed to be created.

are you running as none root?

Megajin commented:
@djsisson I can confirm, that in a non-root install the supabase installation will fail with permission issues.
@retoruto-carry thank you for your fix, I was finally able to install supabase without any issues. Cheers!

peaklabs-dev commented:
Thanks for the PR. But since the current template is working, I do not think we should add this. Also non-root user support is not fully working in many places. 

I think we will get fully working non-root user support with v5, but not sure how it will work at the moment, so I will close this for now.
